examples/linear_deformation.py

```diff
@@ -0,0 +1,1187 @@
+                                            self.discr_space.ndim)
+
+        super().__init__(par_space, self.range_space, linear=True)
+
+        self.ft_kernel = ft_kernel
+
+        if not isinstance(control_points, odl.TensorGrid):
+            self._control_pts = np.asarray(control_points)
+            if self._control_pts.shape != (self.num_contr_pts, self.ndim):
+                raise ValueError(
+                    'expected control point array of shape {}, got {}.'
+                    ''.format((self.num_contr_pts, self.ndim),
+                              self.control_points.shape))
+        else:
+            self._control_pts = control_points
+
+        # TODO: check that number of control points is the same as alphas
+
+    @property
+    def ndim(self):
+        """Number of dimensions of the deformation."""
+        return self.domain.size
+
+    @property
+    def contr_pts_is_grid(self):
+        """`True` if the control points are given as a grid."""
+        return isinstance(self.control_points, odl.TensorGrid)
+
+    @property
+    def control_points(self):
+        """The control points for the deformations."""
+        return self._control_pts
+
+    @property
+    def num_contr_pts(self):
+        """Number of control points for the deformations."""
+        if self.contr_pts_is_grid:
+            return self.control_points.size
+        else:
+            return len(self.control_points)
+
+    @property
+    def image_grid(self):
+        """Spatial sampling grid of the image space."""
+        return discr_space.grid
+
+    def displacement_ft(self, alphas):
+        """Calculate the inverse translation at point y by n-D FFT.
+
+        inverse displacement: v(y)
+
+        Parameters
+        ----------
+        alphas: `ProductSpaceVector`
+            Deformation parameters for control points. It has ``n``
+            components, each of which has size ``N``. Here, ``n`` is
+            the number of dimensions and ``N`` the number of control
+            points. Note that here N = M.
+        kernel: Gaussian kernel fuction
+        """
+        ft_momenta = vectorial_ft_fit_op(alphas)
+        ft_displacement = self.ft_kernel * ft_momenta
+        return vectorial_ft_fit_op.inverse(ft_displacement)
+        # scaling
+#        return (vectorial_ft_op_inverse(ft_displacement) /
+#                self.discr_space.cell_volume * 2.0 * np.pi)
+
+    def _call(self, alphas):
+        """Implementation of ``self(alphas, out)``.
+
+        Parameters
+        ----------
+        alphas: `ProductSpaceVector`
+            Deformation parameters for control points. It has ``n``
+            components, each of which has size ``N``. Here, ``n`` is
+            the number of dimensions and ``N`` the number of control
+            points.
+
+        out : `DiscreteLpVector`
+            Element where the result is stored
+        """
+
+        return self.displacement_ft(alphas)
+
+    def derivative(self, alphas):
```

Comment:
I dont see it. If an operator is linear, it is *by definition* its own derivative. Are you perhaps using some other notion of derivative?